localstack-core/localstack/runtime/runtime.py

```diff
@@ -0,0 +1,213 @@
+import logging
+import os
+import threading
+
+from plux import PluginManager
+
+from localstack import config, constants
+from localstack.runtime import events, hooks
+from localstack.utils import files, net, sync, threads
+
+from .components import Components
+
+LOG = logging.getLogger(__name__)
+
+
+class LocalstackRuntime:
+    """
+    The localstack runtime. It has the following responsibilities:
+
+      - Manage localstack filesystem directories
+      - Execute runtime lifecycle hook plugins from ``localstack.runtime.hooks``.
+      - Manage the localstack SSL certificate
+      - Serve the gateway (It uses a ``RuntimeServer`` to serve a ``Gateway`` instance coming from the
```

Comment:
Q: What's the reason by the way why we don't just keep calling it the `GatewayServer`, like in the (now deprecated) config variable?